tests/functions/length.json

```diff
@@ -52,6 +52,18 @@
       "document" : [{"d": "f"}],
       "result": []
     },
+    {
+      "name": "filter on primitive array element, selects nothing",
+      "selector": "$[?length(@)==1]",
+      "document": [1],
+      "result": []
+    },
+    {
+      "name": "filter on primitive object member value, selects nothing",
```

Comment:
You are right; I mainly added this to make sure no implementation misinterprets `@` as being the whole JSON object (instead of just the member value `1`).

Reply:
I think that can be tested without the `length()` function.

Reply:
Could you provide an example for such a test please?

My intention was that if a library misinterprets `@` as being the array, then `length(@) == 1` passes and it erroneously selects the array as result. Arguably this is not so much about the `length` function.

Maybe using `@==@` could work here though (and this could be in `filter.json` then)? The expected result is `[1]`, but a library which erroneously considers `@` as referring to the JSON array would have `[[]]` or `[[], 1]` or similar as result?

Reply:
`$[?@ == 1]` against either an object or an array works just fine to test this.

Reply:
But that would not detect `@` being misinterpreted as referring also to the array (instead of just its items). It would in all cases only have the `1` as result.

Reply:
> It would in all cases only have the 1 as result.

Yes, exactly, and in doing so, it correctly tests that the implementation is interpreting `@` not as the container but as the value within it.